File: backend/app/schemas/document.py
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID
from app.schemas.common import PaginatedResponse


class DocumentResponse(BaseModel):
    id: UUID
    filename: str
    original_filename: str
    file_type: str
    file_size_bytes: int
    title: str | None = None
    page_count: int | None = None
    chunk_count: int
    status: str
    error_message: str | None = None
    uploaded_at: datetime
    processed_at: datetime | None = None

    model_config = {"from_attributes": True}


class DocumentUploadResponse(BaseModel):
    id: UUID
    filename: str
    status: str
    message: str


class DocumentListResponse(PaginatedResponse):
    documents: list[DocumentResponse]


class DocumentStatusResponse(BaseModel):
    id: UUID
    status: str
    chunk_count: int
    error_message: str | None = None

File: frontend/src/types/document.ts
export interface DocumentResponse {
  id: string;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size_bytes: number;
  title: string | null;
  page_count: number | null;
  chunk_count: number;
  status: "processing" | "ready" | "error";
  error_message: string | null;
  uploaded_at: string;
  processed_at: string | null;
}

export interface DocumentUploadResponse {
  id: string;
  filename: string;
  status: string;
  message: string;
}

export interface DocumentListResponse {
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
  documents: DocumentResponse[];
}

export interface DocumentStatusResponse {
  id: string;
  status: string;
  chunk_count: number;
  error_message: string | null;
}
